var/spack/repos/builtin/packages/tau/package.py

```diff
@@ -131,6 +133,9 @@ def set_compiler_options(self, spec):
         compiler_options.append(useropt)
         return compiler_options
 
+    def setup_build_environment(self, env):
+        env.prepend_path('LIBRARY_PATH', self.spec['zlib'].prefix.lib)
```

Comment:
This is still needed even with the link dep on zlib?

Reply:
Yes. I double checked after changing it to a link dep and it still breaks without the build setup (in an environment that doesn't have zlib in a standard system directroy). 

It's possible something will come up with that elf dependency again but I wasn't able to reproduce the issue I thought I saw with it last week.